src/core/HY_Features.cpp

```diff
@@ -0,0 +1,61 @@
+#include <HY_Features.hpp>
+#include <HY_PointHydroNexus.hpp>
+
+using namespace hy_features;
+
+HY_Features::HY_Features( geojson::GeoJSON fabric, std::shared_ptr<Formulation_Manager> formulations)
+  :HY_Features(network::Network(fabric), formulations)
+{
+  /* If you nest the constuctor calls like this, then the network, while valid in the next constructor,
+     is empty once the collection is fully instansiated...
+    HY_Features(network::Network(fabric));
+  */
+}
+
+HY_Features::HY_Features(network::Network network, std::shared_ptr<Formulation_Manager> formulations)
+  :network(network), formulations(formulations)
+{
+      std::string feat_id;
+      std::string feat_type;
+      std::vector<std::string> origins, destinations;
+
+      for(const auto& feat_idx : network){
+        feat_id = network.get_id(feat_idx);//feature->get_id();
+        feat_type = feat_id.substr(0, 3);
+
+        destinations  = network.get_destination_ids(feat_id);
+        if(feat_type == "cat")
```

Comment:
I wish we had a way to get away from having different logic based on the first three characters of an ID, but that a discussion for another day.